frontend/src/components/explorer/interactionPartners/DetailsPage.vue

```diff
@@ -536,10 +536,13 @@ export default {
       // map appears "flat".
       this.controller.setCamera({ x: 0, y: 0, z: lz });
     },
-    async applyColors() {
+    async applyColors({ deselect } = { deselect: false }) {
       if (this.controller) {
         const colors = {};
-        this.network.nodes.forEach(node => {
+        const nodes = deselect
+          ? this.network.nodes.filter(n => n.id === this.clickedElmId)
```

Comment:
Ah this was a much cleaner way of obtaining the required node data compared to what I had in mind, nice!